src/azure-cli/azure/cli/command_modules/role/custom.py

```diff
@@ -99,7 +99,7 @@ def _create_update_role_definition(cmd, role_definition, for_update):
         if not role_name:
             raise CLIError("please provide role name")
 
-    if not for_update and 'assignableScopes' not in role_definition:
+    if not for_update and not('assignableScopes' in role_definition and role_definition['assignableScopes']):
```

Comment:
```suggestion
    if not for_update and not(role_definition.get('assignableScopes')):
```

Reply:
Good point. `assignableScopes` can actually be omitted, use `role_definition.get('assignableScopes', None)` instead.